cmd/owner.go

```diff
@@ -213,6 +215,90 @@ func getOwnerServerState(config *OwnerServerConfig) (*OwnerServerState, error) {
 	}, nil
 }
 
+func validateFSIMParameters() error {
+	// Only validate if FSIM parameters are actually being used
+	if !hasFSIMParameters() {
+		return nil // No FSIM parameters to validate
+	}
+
+	// Parse and validate wget URLs
+	wgetURLs = make([]*url.URL, 0, len(wgets))
+	for _, urlString := range wgets {
+		parsedURL, err := url.Parse(urlString)
+		if err != nil {
+			return fmt.Errorf("invalid wget URL %q: %w", urlString, err)
+		}
+		if parsedURL.Scheme != "http" && parsedURL.Scheme != "https" {
+			return fmt.Errorf("wget URL %q must use http or https scheme, got %q", urlString, parsedURL.Scheme)
+		}
+		if parsedURL.Host == "" {
+			return fmt.Errorf("wget URL %q missing host", urlString)
+		}
+		wgetURLs = append(wgetURLs, parsedURL)
+	}
+
+	// Validate and store cleaned download file paths
+	downloadPaths = make([]string, 0, len(downloads))
+	for _, filePath := range downloads {
+		cleanPath := filepath.Clean(filePath)
+		file, err := os.Open(cleanPath)
+		if err != nil {
+			if os.IsNotExist(err) {
+				return fmt.Errorf("download file %q does not exist", filePath)
+			}
+			if os.IsPermission(err) {
+				return fmt.Errorf("download file %q is not readable", filePath)
+			}
+			return fmt.Errorf("cannot access download file %q: %w", filePath, err)
+		}
+		if err := file.Close(); err != nil {
+			return fmt.Errorf("error closing download file %q: %w", filePath, err)
+		}
+		downloadPaths = append(downloadPaths, cleanPath)
+	}
+
+	if len(uploads) > 0 && uploadDir == "" {
+		return fmt.Errorf("upload directory must be specified when using --command-upload")
+	}
+
+	if uploadDir != "" {
+		info, err := os.Stat(uploadDir)
+		if err != nil {
+			if os.IsNotExist(err) {
+				return fmt.Errorf("upload directory %q does not exist", uploadDir)
+			}
```

Comment:
you can drop this check, the error gives you already the context like "file not found"

Reply:
Removed in update